### PR TITLE
Story 2 - Display single book

### PR DESCRIPTION
**Pull request by GDTimi**
Story 2 work, submitted for a review please!

- As a person who likes to read
- I want to see all the details about a specific book
- So that I can decide whether I want to read it

Clicking a book displays its details on a new page
title, blurb, image, page count, year (to be confirmed), author, & genre
The title in the navbar should act as a link back to the homepage
Follow the wireframe provided

### CODE REVIEW

File: src/components/BookDetail/BookDetail.jsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+                setBookAuthor(bookData.data.author)
+                setBookYear(bookData.data.year)
+                setBookPageCount(bookData.data.page_count)
+                setBookGenre(bookData.data.genre.name)
+                setBookBlurb(bookData.data.blurb)
+            } else {
+                // Display an error message
+                setError(true)
+            }
+
```
Comment (Andrei-Trifu) [inline src/components/BookDetail/BookDetail.jsx]: maybe remove some of these spaces

File: src/components/BookDetail/BookDetail.jsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+        })
+
```
Comment (Andrei-Trifu) [inline src/components/BookDetail/BookDetail.jsx]: this one too

**Review by Andrei-Trifu — APPROVED** [review]
Looks good to me

File: src/Components/HomePage/AllBooks/AllBooks.jsx [hunk 3]
```diff
@@ -1,35 +1,44 @@
...
+                    <NavLink to={"/book/" + bookData.id}>
+                    <BookListing
+                        key={bookData.id}
+                        id={bookData.id}
+                        title={bookData.title}
+                        author={bookData.author}
+                        image={bookData.image}
+                        genre={bookData.genre.name}
+                        setClickedBook ={setClickedBook}
+                    />
```
Comment (Geoff10) [inline src/Components/HomePage/AllBooks/AllBooks.jsx]: Could this bit of indentation be tidied up, please?

File: src/Components/HomePage/AllBooks/AllBooks.jsx [hunk 4]
```diff
@@ -1,35 +1,44 @@
...
 import BookListing from "./BookListing/BookListing"
 import AllBooks from "./AllBooks.css"
 
 function Allbooks() {
     
 const [allBooks, setAllBooks] = useState([])
 
+//Contains the ID of the book that the user selects
+const[clickedBook, setClickedBook] = useState(null);
+
```
Comment (Geoff10) [inline src/Components/HomePage/AllBooks/AllBooks.jsx]: The indentation isn't quite right here, the constants are a part of your `AllBooks` compomnent.

File: src/Components/HomePage/AllBooks/AllBooks.jsx [hunk 5]
```diff
@@ -1,35 +1,44 @@
 import { useEffect, useState } from "react"
+import { NavLink } from "react-router-dom";
 import BookListing from "./BookListing/BookListing"
 import AllBooks from "./AllBooks.css"
 
 function Allbooks() {
```
Comment (Geoff10) [inline src/Components/HomePage/AllBooks/AllBooks.jsx]: The casing is inconsistent on this component name, it should also have a captial 'B'

File: src/Components/HomePage/AllBooks/AllBooks.jsx [hunk 6]
```diff
@@ -1,35 +1,44 @@
 import { useEffect, useState } from "react"
+import { NavLink } from "react-router-dom";
 import BookListing from "./BookListing/BookListing"
 import AllBooks from "./AllBooks.css"
```
Comment (Geoff10) [inline src/Components/HomePage/AllBooks/AllBooks.jsx]: When importing the CSS you do not need to name the import, `"./AllBooks.css"` will work and also fix a warning being output in your console.

File: src/Components/HomePage/AllBooks/BookListing/BookListing.jsx [hunk 7]
```diff
@@ -1,19 +1,15 @@
 import "./BookListing.css"
 
-
-function BookListing({image, title, author, genre,}) {
-
+function BookListing({id, image, title, author, genre,}) {
```
Comment (Geoff10) [inline src/Components/HomePage/AllBooks/BookListing/BookListing.jsx]: The `id` variable isn't being used.

File: src/components/BookDetail/BookDetail.css [hunk 8]
```diff
@@ -0,0 +1,60 @@
+@import url('https://fonts.googleapis.com/css2?family=Pixelify+Sans&family=Sometype+Mono&display=swap');
```
Comment (Geoff10) [inline src/components/BookDetail/BookDetail.css:1]: Can this file be tidied up for consistency, please? Mainly, decide if you are you going to be putting spaces after CSS declaration blocks are not and then update this file accordingly.

File: src/components/BookDetail/BookDetail.jsx [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+                setBookPageCount(bookData.data.page_count)
+                setBookGenre(bookData.data.genre.name)
+                setBookBlurb(bookData.data.blurb)
+            } else {
+                // Display an error message
+                setError(true)
+            }
+
+
+        })
```
Comment (Geoff10) [inline src/components/BookDetail/BookDetail.jsx]: The indentation is not quite right here, the fetch is inside the useEffect.

**Review by Geoff10 — APPROVED** [review]
Really good work, a few small comments. Also, I noticed that the navbar contains links that are not specified in the wireframe, these links also don't work at the moment.

File: src/Components/HomePage/AllBooks/AllBooks.jsx (hunk 4 above)
Comment (GDTimi) [reply src/Components/HomePage/AllBooks/AllBooks.jsx]: Yep, corrected.

File: src/Components/HomePage/AllBooks/AllBooks.jsx (hunk 3 above)
Comment (GDTimi) [reply src/Components/HomePage/AllBooks/AllBooks.jsx]: Fixed!

File: src/Components/HomePage/AllBooks/AllBooks.jsx (hunk 5 above)
Comment (GDTimi) [reply src/Components/HomePage/AllBooks/AllBooks.jsx]: Cheers, fixed

File: src/Components/HomePage/AllBooks/AllBooks.jsx (hunk 6 above)
Comment (GDTimi) [reply src/Components/HomePage/AllBooks/AllBooks.jsx]: Fixed!

File: src/Components/HomePage/AllBooks/BookListing/BookListing.jsx (hunk 7 above)
Comment (GDTimi) [reply src/Components/HomePage/AllBooks/BookListing/BookListing.jsx]: Removed

File: src/components/BookDetail/BookDetail.css (hunk 8 above)
Comment (GDTimi) [reply src/components/BookDetail/BookDetail.css:1]: Thanks, spacing fixed.

File: src/components/BookDetail/BookDetail.jsx (hunk 9 above)
Comment (GDTimi) [reply src/components/BookDetail/BookDetail.jsx]: Fixed!

**Review by Geoff10 — APPROVED** [review]

**Pull request merged**